src/app/components/pages/admin-cell/admin-cell.component.ts

```diff
@@ -315,10 +343,147 @@ export class AdminCellComponent implements OnInit {
+              apiError: err.error.nb_volunteers_needed
+            });
+          }
+          if (err.error.cycle_id) {
+            this.eventForm.controls['cycle_id'].setErrors({
+              apiError: err.error.cycle_id
+            });
+          }
+          if (err.error.task_type_id) {
+            this.eventForm.controls['task_type_id'].setErrors({
+              apiError: err.error.task_type_id
+            });
+          }
+          if (err.error.start_date) {
+            this.eventForm.controls['start_date'].setErrors({
+              apiError: err.error.start_date
+            });
+          }
+          if (err.error.end_date) {
+            this.eventForm.controls['end_date'].setErrors({
+              apiError: err.error.end_date
+            });
+          }
+          if (err.error.non_field_errors) {
+            this.eventForm.setErrors({
+              apiError: err.error.non_field_errors
+            });
+          }
+        }
+      );
+  }
+
+  opEvent() {
+
+    if (this.eventForm.valid) {
+      if (this.modalCreate) {
+        this.createEvent();
+      } else {
+        this.updateEvent(this.modalEventId);
+      }
+
     } else {
       for (const controlKey of Object.keys(this.eventForm.controls)) {
         this.eventForm.controls[controlKey].markAsTouched();
       }
     }
   }
+
+  getVolunteersField(event: Event) {
+    return event.nb_volunteers + ' / ' + event.nb_volunteers_needed;
+  }
+  getStandByField(event: Event) {
+    return event.nb_volunteers_standby + ' / ' + event.nb_volunteers_standby_needed;
+  }
+
+  getDateField(strDate: string) {
+    const formatedDate = new Date(strDate);
+
+    const options = {
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric' };
+
+    return formatedDate.toLocaleDateString('fr', options) + ' ' +
+      ('0' + formatedDate.getHours()).slice(-2) + ':' +
+      ('0' + formatedDate.getMinutes()).slice(-2);
+  }
+
+  getAddressField(address: Address) {
+    const arrayStr = [
+      address.address_line1,
+      address.address_line2,
+      address.postal_code + ' ' + address.city
+    ];
+
+    return arrayStr;
+  }
+
+  eventsAdapter() {
+    const eventsAdapted = [];
+    for (const event of this.events) {
+      const newEvent = {
+        id: event.id,
+        start_date: this.getDateField(event.start_date),
+        end_date: this.getDateField(event.end_date),
+        task_type: event.task_type.name,
+        volunteers: this.getVolunteersField(event),
+        stand_by: this.getStandByField(event),
+        model: event
+      };
+      eventsAdapted.push(newEvent);
+    }
+    return eventsAdapted;
+  }
+
+  openModalCreateEvent() {
+    this.modalTitle = 'Création d\'une plage horaire';
+    this.modalCreate = true;
+    this.toogleModal();
+  }
+
+  openModalEditEvent(event) {
+
+    this.modalTitle = 'Modification d\'une plage horaire';
+    this.modalCreate = false;
+    this.modalEventId = event.model.id;
```

Comment:
Since `modalEventId` is always defined when we use the modal to update we can maybe get rid of `modalCreate` and juste use one attribute

Reply:
Good idea !